dcplib/etl/__init__.py

```diff
@@ -0,0 +1,128 @@
+"""
+Shared ETL (extract, transform, load) code for fetching and loading data from HCA DSS.
+
+The main entry point is the extract() function, which takes optional transform and load callbacks that will be invoked
+for each bundle fetched.
+"""
+
+import os, sys, json, concurrent.futures, hashlib, logging, threading
+from collections import defaultdict
+from fnmatch import fnmatchcase
+
+import hca
+from ..networking import http
+from hca.util import RetryPolicy
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+class DSSExtractor:
+    default_bundle_query = {'query': {'bool': {'must_not': {'term': {'admin_deleted': True}}}}}
+    default_content_type_patterns = ['application/json; dcp-type="metadata*"']
+
+    def __init__(self, staging_directory):
+        self.dss_host = hca.dss.DSSClient().host
```

Comment:
Assuming DSS environment will be set via CLI config. Will you add an option to set this via the script?

Reply:
Fixed (now in the init args)